internal/store/googlesm/google_secret_manager.go

```diff
@@ -0,0 +1,158 @@
+package googlesm
+
+import (
+	"context"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"log/slog"
+	"regexp"
+
+	"google.golang.org/api/iterator"
+
+	"github.com/Layr-Labs/bn254-keystore-go/curve"
+	"github.com/Layr-Labs/bn254-keystore-go/keystore"
+
+	"github.com/Layr-Labs/cerberus/internal/crypto"
+	"github.com/Layr-Labs/cerberus/internal/store"
+
+	secretmanager "cloud.google.com/go/secretmanager/apiv1"
+	"cloud.google.com/go/secretmanager/apiv1/secretmanagerpb"
+)
+
+var _ store.Store = (*Keystore)(nil)
+
+const (
+	storagePrefix = "cerberus"
+	ProjectKey    = "project"
+)
+
+type Keystore struct {
+	smClient  *secretmanager.Client
+	projectID string
+
+	logger *slog.Logger
+}
+
+func NewKeystore(projectID string, logger *slog.Logger) (*Keystore, error) {
+	ctx := context.Background()
+	client, err := secretmanager.NewClient(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Keystore{
+		smClient:  client,
+		projectID: projectID,
+		logger:    logger.With("component", "google-secret-manager-store"),
+	}, nil
+}
+
+func (k Keystore) RetrieveKey(
+	ctx context.Context,
+	pubKey string,
+	password string,
+) (*crypto.KeyPair, error) {
+	// Build the request
+	secretID := storagePrefix + pubKey
+	accessRequest := &secretmanagerpb.AccessSecretVersionRequest{
+		Name: fmt.Sprintf("projects/%s/secrets/%s/versions/latest", k.projectID, secretID),
+	}
+
+	// Access the secret version
+	result, err := k.smClient.AccessSecretVersion(ctx, accessRequest)
+	if err != nil {
+		return nil, fmt.Errorf("failed to access secret version: %v", err)
+	}
+
+	privKeyHex := hex.EncodeToString(result.Payload.Data)
+	kp, err := crypto.NewKeyPairFromHexString(privKeyHex)
+	if err != nil {
+		return nil, err
+	}
+
+	return kp, nil
+}
+
+func (k Keystore) StoreKey(ctx context.Context, keyPair *keystore.KeyPair) (string, error) {
+	pubKey, err := keystore.BlsSkToPk(keyPair.PrivateKey, string(curve.BN254))
+	if err != nil {
+		return "", err
+	}
+
+	storageKey := storagePrefix + pubKey
+
+	createSecretReq := &secretmanagerpb.CreateSecretRequest{
+		Parent:   fmt.Sprintf("projects/%s", k.projectID),
+		SecretId: storageKey,
+		Secret: &secretmanagerpb.Secret{
+			Labels: map[string]string{
+				ProjectKey: storagePrefix,
+			},
+			Replication: &secretmanagerpb.Replication{
+				Replication: &secretmanagerpb.Replication_Automatic_{
+					Automatic: &secretmanagerpb.Replication_Automatic{},
+				},
+			},
```

Comment:
What does this do exactly? Is it something we want turned on by default?

Reply:
it replicates the key in multiple zones. Given they are keys, I think it's good to do by default.